tokio/src/time/timeout.rs

```diff
@@ -45,14 +48,20 @@ use std::task::{self, Poll};
 /// }
 /// # }
 /// ```
+#[cfg_attr(tokio_track_caller, track_caller)]
 pub fn timeout<T>(duration: Duration, future: T) -> Timeout<T>
 where
     T: Future,
 {
+    #[cfg(all(tokio_track_caller, tokio_unstable, feature = "tracing"))]
+    let location = Some(Location::caller());
+    #[cfg(not(all(tokio_track_caller, tokio_unstable, feature = "tracing")))]
+    let location = None;
```

Comment:
if we made a wrapper function for getting the callsite location, as i suggested above, this could also just be
```suggestion
    let location = crate::util::trace::caller_location();
```